### PR TITLE
[refs #285] Add contain modifier

### PR DESCRIPTION
**Pull request by florianbouvot**
Does this modifier fit framework needs ?

### CODE REVIEW

File: objects/_objects.ratio.scss [hunk 1]
```diff
@@ -83,3 +83,26 @@ $inuit-ratios: (
...
+
+
+/* Contain modifiers.
+   ========================================================================== */
+
+/**
+ * Only works with image content.
+ */
+
+.o-ratio--contain {
```
Comment (anselmh) [inline objects/_objects.ratio.scss]: As it only works for image content, we should probably match the name of the class as well.

**Review by anselmh — CHANGES_REQUESTED** [review]

File: objects/_objects.ratio.scss (hunk 1 above)
Comment (csshugs) [reply objects/_objects.ratio.scss]: Second that. Something like `.o-ratio--img-contain` would be more concise.

**Review by csshugs — CHANGES_REQUESTED** [review]
@florianbouvot 
Since #309 is merged, this PR also has to provide a proper [changelog entry](https://github.com/inuitcss/inuitcss/blob/117df172d8edf38998e8ea643272533bee7261ac/CHANGELOG.md).

File: objects/_objects.ratio.scss [hunk 2]
```diff
@@ -83,3 +83,26 @@ $inuit-ratios: (
 }
 
 /* stylelint-enable */
+
+
+
+
+
```
Comment (olivertappin) [inline objects/_objects.ratio.scss]: Do you need this many line breaks? Would be nice to keep it consistent and have only 3 (rather than 5).

Comment (csshugs) [reply objects/_objects.ratio.scss]: @olivertappin Actually, 5 empty lines above new sections in a file is a consistent convention across the framework. See [media object](https://github.com/inuitcss/inuitcss/blob/29ec23cfdfc5e8004f31e41a4c9f701cfa0ba642/objects/_objects.media.scss) for example.

**Comment by ArmorDarks** [comment]
I wonder what exactly does it do?

**Comment by csshugs** [comment]
@ArmorDarks 
See #285.

Basically, it mimics the `object-fit: contain` declaration.
([Further reading](https://css-tricks.com/almanac/properties/o/object-fit/))

File: objects/_objects.ratio.scss (hunk 1 above)
Comment (florianbouvot) [reply objects/_objects.ratio.scss]: @anselmh @csshugs you're right, I'll update that today.

**Comment by rowild** [comment]
Is it "contain" or "contains"???
And perhaps a plural on modifier: "modifier(s)"?

**Comment by herzinger** [comment]
Just look at the code, if you don't understand it. It's a single modifier, so why would it be plural? Also, it's "contain", as in `background-size: contain`, a css property with a similar behavior.

**Comment by rowild** [comment]
@herzinger no need to be so harsh. 
Grammatically seen, it is wrong. We are talking about a comment: it starts with a capital letter and the sentence ends with a dot. The way I understand it is "The following sections contains modifiers."
```
// Contain modifiers.
```
But this is simply wrong, the correct spelling would be "Contains modifiers."
If you want to point out that, what follows, is subject to the css property "contain" - which is clearly what you want here -, then this spelling would be correct(er): 
```
// 'contain' modifiers.
```
(or something similar that takes out "contain" from the sentence grammar).
I know this is nit picking. But from what I read in other conversations I am not the only one, and since my profession is proof-reading (although for German, not for English), I thought I might contribute very carefully (using question marks). But since you are so sure about what you do, I better refrain from that in future.
Fazit remaining: The way it is written, is wrong, at least unclear. And wrong/unclear spelling rises the question, whether the code might be wrong, too. (A "rule of consequence" that was NOT invented by me...)

**Comment by herzinger** [comment]
@rowild sorry if it sounded harsh, that wasn't my intention. It wasn't also directed at you specifically, hence the lack of a mention. Still, what I said still applies, so let me me more verbose. 

I had no way of knowing exactly what you were referring to in the original comment, since it wasn't in code.  Even knowing it's about the section title-like comment you pointed out now, it's a single modifier, so it's correct as singular. Also, "Contain" is the name of said modifier, so it's here not as a verb, but a substantive. While I agree that quotation marks could make it clearer, the way it is right now is the same used in other pieces of code in the framework (like the [layout object](https://github.com/inuitcss/inuitcss/blob/develop/objects/_objects.layout.scss), for instance), and it is grammatically correct. 

```
If you want to point out that, what follows, is subject to the css property "contain" - which is clearly what you want here
```

It's not. The NAMING of the modifier is inspired by css declarations like `background-size: contain` and `object-fit: contain` (like @csshugs pointed out), since the intended behavior is similar in many ways, but only that. It's still a BEM-like modifier, the same as all other modifiers across the framework. My main point is that it would become clear by reading the document as a whole, instead of looking for meaning in this subtitle alone.

That out of the way, there is one correction and one addition that, as I see, need to be made:

1. It's currently pluralized, and as I said above, it's a single modifier, so the correct wold be singular: 
```Contain modifier```
2. There should be a short description of the effect expected by applying this modifier, similar to what you see in the other objects. Something to the effect of: 
```
Only works with image content.
Scales the image as large as possible 
without cropping or stretching it.
```

**Comment by florianbouvot** [comment]
@csshugs changelog is not present in this branch, what should I do?

**Comment by csshugs** [comment]
@florianbouvot 
The changelog is already in the `develop` branch, so you could merge `develop` into this feature branch.

**Comment by csshugs** [comment]
@florianbouvot Tell me if you need help with this or don't have the time currently. I could easily take over the changelog part, if you like.

**Review by csshugs — APPROVED** [review]

**Review by anselmh — APPROVED** [review]

**Comment by lennerd** [comment]
@florianbouvot Maybe I'm blind or anything, but I cannot really see why you use `.o-ratio__content:before`. Isn‘t this selector missing a content statement?

**Pull request merged**